### PR TITLE
chore: deprecate gauge metrics with _total suffix (#12744)

### LINKED ISSUE #12744: Deprecate gauge metrics named like counters
`coderd_oauth2_external_requests_rate_limit_total` and `coderd_api_workspace_latest_build_total` are both gauges, but use the `_total` suffix which is usually used to indicate a counter, as described by the prometheus [documentation on naming](https://prometheus.io/docs/instrumenting/writing_exporters/#naming:~:text=_total%20is%20a%20convention%20for%20counters%2C%20you%20should%20use%20it%20if%20you%E2%80%99re%20using%20the%20COUNTER%20type.).

While this is not strictly a problem, it may lead to users incorrectly assuming they can perform aggregations on these metrics rather than just using their current value; aggregations such as `rate` on gauges make no sense.

I'd suggest that we deprecate these metric names, export another metric without the `_total` suffix alongside them, and mark the original metrics for removal in the next major release. _IMHO metrics are part of the API surface, and therefore should be treated in a semver-observant manner._

### PR DESCRIPTION
**Pull request by snark87**
Fixes coder/coder#12744

Deprecated metrics:
- coderd_oauth2_external_requests_rate_limit_total
- coderd_api_workspace_latest_build_total

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]
Great work @snark87! Thank you very much for your contribution

File: coderd/prometheusmetrics/prometheusmetrics.go [hunk 1]
```diff
@@ -79,10 +79,22 @@ func Workspaces(ctx context.Context, logger slog.Logger, registerer prometheus.R
 		duration = defaultRefreshRate
 	}
 
-	workspaceLatestBuildTotals := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	// TODO: deprecated: remove in the future
```
Comment (johnstcn) [inline coderd/prometheusmetrics/prometheusmetrics.go:82]:
```suggestion
	// TODO: deprecated: remove in the future
	// See: https://github.com/coder/coder/issues/12999
```

File: coderd/promoauth/oauth2.go [hunk 2]
```diff
@@ -91,14 +93,25 @@ func NewFactory(registry prometheus.Registerer) *Factory {
...
-				Name:      "external_requests_rate_limit_total",
+				Name:      "external_requests_rate_limit",
 				Help:      "The total number of allowed requests per interval.",
 			}, []string{
 				"name",
 				// Resource allows different rate limits for the same oauth2 provider.
 				// Some IDPs have different buckets for different rate limits.
 				"resource",
 			}),
+			// TODO: deprecated: remove in the future
```
Comment (johnstcn) [inline coderd/promoauth/oauth2.go:104]:
```suggestion
			// TODO: deprecated: remove in the future
			// See: https://github.com/coder/coder/issues/12999
```

**Review by johnstcn — APPROVED** [review]
Nice work! Created a follow-up issue to track deprecation.

**Comment by johnstcn** [comment]
@snark87 can you please merge latest main? There's an issue with one of the tests that has been worked around here: https://github.com/coder/coder/issues/12978

**Comment by snark87** [comment]
> @snark87 can you please merge latest main? There's an issue with one of the tests that has been worked around here: #12978
Done, CI is green now.

**Pull request merged**